src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -173,48 +173,86 @@ protected override Conversion GetInterpolatedStringConversion(BoundExpression so
                     break;
             }
 
-            Debug.Assert(collectionTypeKind == CollectionExpressionTypeKind.CollectionInitializer || elementType is { });
-
-            if (collectionTypeKind == CollectionExpressionTypeKind.CollectionInitializer)
+            var elements = node.Elements;
+            if (collectionTypeKind == CollectionExpressionTypeKind.IEnumerable)
+            {
+                return Conversion.CreateCollectionExpressionConversion(collectionTypeKind, elementType: null, default);
+            }
+            else if (collectionTypeKind == CollectionExpressionTypeKind.GenericIEnumerable)
             {
-                var implicitReceiver = new BoundObjectOrCollectionValuePlaceholder(syntax, isNewInstance: true, targetType) { WasCompilerGenerated = true };
-                var diagnostics = BindingDiagnosticBag.Discarded;
-                var collectionInitializerAddMethodBinder = _binder.WithAdditionalFlags(BinderFlags.CollectionInitializerAddMethod);
-                foreach (var element in node.Elements)
+                ImmutableArray<NamedTypeSymbol> allInterfaces;
+                switch (targetType.TypeKind)
                 {
-                    _ = _binder.BindCollectionExpressionElementAddMethod(
-                        element,
-                        collectionInitializerAddMethodBinder,
-                        implicitReceiver,
-                        diagnostics,
-                        out bool hasErrors);
-                    if (hasErrors)
+                    case TypeKind.Class:
+                    case TypeKind.Struct:
+                        allInterfaces = targetType.AllInterfacesNoUseSiteDiagnostics;
+                        break;
+                    case TypeKind.TypeParameter:
+                        allInterfaces = ((TypeParameterSymbol)targetType).AllEffectiveInterfacesNoUseSiteDiagnostics;
+                        break;
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(targetType.TypeKind);
+                }
+
+                var ienumerableType = this.Compilation.GetSpecialType(SpecialType.System_Collections_Generic_IEnumerable_T);
+                foreach (var @interface in allInterfaces)
+                {
+                    if (isCompatibleGenericIEnumerable(@interface, ienumerableType, elements, ref useSiteInfo))
```

Comment:
>useSiteInfo [](http://example.com/codeflow?start=98&length=11)

If we're combining the use-site diagnostics from multiple `IEnumerable<T>` implementations, it feels like we should call `isCompatibleGenericIEnumerable()` on all of the implementations, rather than stopping at the first compatible implementation.